tests/mock_api/test_continuous_mastering.py

```diff
@@ -63,3 +63,62 @@ def test_continuous_mastering():
 
     clause1 = project.estimate_pairs().clause_estimates["clause1"]
     assert clause1["generatedPairCount"] == "25"
+
+
+@responses.activate
+def test_record_clusters_with_data():
+
+    project_config = {
+        "name": "Project 1",
+        "description": "Mastering Project",
+        "type": "DEDUP",
+        "unifiedDatasetName": "Project 1 - Unified Dataset",
+        "externalId": "Project1",
+        "resourceId": "1",
+    }
+
+    unified_dataset_json = {
+        "id": "unify://unified-data/v1/datasets/8",
+        "name": "Project_1_unified_dataset",
+        "version": "10",
+        "relativeId": "datasets/8",
+        "externalId": "Project_1_unified_dataset",
+    }
+
+    datasets_json = [
+        {
+            "id": "unify://unified-data/v1/datasets/36",
+            "name": "Project_1_unified_dataset_dedup_clusters_with_data",
+            "version": "251",
+            "relativeId": "datasets/36",
+            "externalId": "1",
+        }
+    ]
+
+    rcwd_json = {
+        "externalId": "1",
+        "id": "unify://unified-data/v1/datasets/36",
+        "name": "Project_1_unified_dataset_dedup_clusters_with_data",
+        "relativeId": "datasets/36",
+        "version": "251",
+    }
+
+    auth = UsernamePasswordAuth("username", "password")
+    unify = Client(auth)
+
+    project_external_id = "1"
+
+    projects_url = (
+        f"http://localhost:9100/api/versioned/v1/projects/{project_external_id}"
+    )
+    unified_dataset_url = (
+        f"http://localhost:9100/api/versioned/v1/projects/1/unifiedDataset"
+    )
+    datasets_url = f"http://localhost:9100/api/versioned/v1/datasets"
+
+    responses.add(responses.GET, projects_url, json=project_config)
+    responses.add(responses.GET, unified_dataset_url, json=unified_dataset_json)
+    responses.add(responses.GET, datasets_url, json=datasets_json)
+    project = unify.projects.by_resource_id("1")
+    actual_rcwd_dataset = project.as_mastering().record_clusters_with_data()
+    assert actual_rcwd_dataset._data == rcwd_json
```

Comment:
By virtue of the leading underscore in its name, `_data` is private and not part of the contract of the Dataset class.
This must instead test using the public interface:
```
assert actual.name == rcwd_json['name']
```
etc.